### PR TITLE
Improve build

### PR DESCRIPTION
**Pull request by Cat-Ion**
- Aborts `build.sh` on errors
- Adds a default value for `IDF_PATH` to `firmware/Makefile`
- Removes `set_env.sh`

### CODE REVIEW

**Review by kliment — DISMISSED** [review]
I'm okay with this

File: firmware/Makefile [hunk 1]
```diff
@@ -1,4 +1,5 @@
 PROJECT_NAME := firmware
 
+IDF_PATH ?= ../esp-idf
```
Comment (Cat-Ion) [inline firmware/Makefile:3]: Small comment here - `setenv.sh` also adds other environment variables, but `make` finishes without errors for me with just this (and the correct `PATH` with the espressif stuff), so I don't know if we need the rest.

Comment (renzenicolai) [reply firmware/Makefile:3]: Love it, you can probably get rid of the set_env.sh entirely indeed.

Comment (Cat-Ion) [reply firmware/Makefile:3]: There you go ;)

**Review by renzenicolai — APPROVED** [review]

**Pull request merged**